rfd/0222-access-list-templates.md

```diff
@@ -0,0 +1,258 @@
+- requester: a role assigned to member grants that allow them to search and request for resources defined in `access` role
+
+##### System roles for long-term template
+
+- access: a role that defines the access to resources and is assigned to member grants
+
+##### Naming system roles
+
+In order to ensure uniqueness and help identifying which roles belong to access lists, the naming convention takes the following format:
+
+`<purpose>-acl-role-<access list metadata name (UID)>`
+
+| Parts                             |                      Explanation                      |              Example Values |
+| :-------------------------------- | :---------------------------------------------------: | --------------------------: |
+| \<purpose\>                       |     short word that describes the purpose of role     | requester, reviewer, access |
+| acl-role                          |             stands for "access list role"             |                         n/a |
+| <access list metadata name (UID)> | helps identify which access list this role belongs to |                         n/a |
+
+Example names of system roles, if an access list with metadata.name is set to `abcd1234`
+
+- requester-acl-role-abcd1234
+- reviewer-acl-role-abcd1234
+- access-acl-role-abcd1234
+
+#### Update
+
+The `type: template` field cannot be modified.
+
+Minus the fields that are already modifiable, `template_config` field is partly modifiable. The field that defines access to resources can be modified. The `oneof type` will not be modifiable eg: if an access list template was originally `short-term`, a user cannot change the template to `long-term`.
+
+For both templates, since only the `access` part of the `template_config` can be modified, the system role related to `access` will be updated.
+
+##### Update quirk
+
+There is a quirk where the `access` definition set on an access list might not be in sync with the actual role resource because we don't prevent users from editing system roles with `tctl` (the web UI does not allow reading/updating system roles). If such a case happens, ultimately the role resource is the source of truth. Any updates made to `template_config` will overwrite any previous edits directly made to the system roles.
+
+#### Delete
+
+In the backend, after an access list is successfully deleted, all system roles tied to that deleted access list will also be deleted.
+
+In the case deleting roles fail for some reason, we can offer a retry if we detect the failure was due to clean up. An API endpoint will be created that is specific to cleaning up templated access list (which is to just delete roles at this moment).
+
+### Phases
+
+#### Phase 1
+
+Add CRUD support for templated access list (both short and long term) through the web UI and tctl.
+
+#### Phase 2
+
+Support scaling templated access list with terraform. In addition, add scaling directions on the web UI as `next steps` after creating an access list is successful.
```

Comment:
i wasn't sure if I can punt support for terraform? (i saw access list member support added later, so that's where i got my assumption from)